resolve/matrix/Coo.cpp

```diff
@@ -308,15 +318,22 @@ namespace ReSolve
     switch (memspace) {
       case HOST:
         if (h_data_updated_) {
-          out::misc() << "In Coo::syncData trying to sync host, but host already up to date!\n";
+          out::misc() << "WARNING: In Coo::syncData trying to sync host, but host already up to date! This line is ignored. (Perhaps you meant to sync device)\n";
           return 0;
```

Comment:
Use built-in warning and also break too long print line.
```suggestion
          out::warning() << In Coo::syncData trying to sync host, but host already up to date!
                         << This line is ignored. (Perhaps you meant to sync device)\n";
          return 0;
```
Please make the same change for similar instances in the PR.